pkg/iac/terraform/state/backend/tfcloud_reader.go

```diff
@@ -26,38 +28,79 @@ type TFCloudBody struct {
 	Data TFCloudData `json:"data"`
 }
 
-func NewTFCloudReader(client pkghttp.HTTPClient, workspaceId string, opts *Options) (*HTTPBackend, error) {
-	req, err := http.NewRequest(http.MethodGet, fmt.Sprintf("%s/workspaces/%s/current-state-version", TFCloudAPI, workspaceId), nil)
-
-	if err != nil {
-		return nil, err
-	}
-
-	req.Header.Add("Content-Type", "application/vnd.api+json")
-	req.Header.Add("Authorization", fmt.Sprintf("Bearer %s", opts.TFCloudToken))
-
-	res, err := client.Do(req)
+type TFCloudBackend struct {
+	request *http.Request
+	client  pkghttp.HTTPClient
+	reader  io.ReadCloser
+}
 
+func NewTFCloudReader(client pkghttp.HTTPClient, workspaceId string, opts *Options) (*TFCloudBackend, error) {
+	req, err := http.NewRequest(http.MethodGet, fmt.Sprintf("%s/workspaces/%s/current-state-version", TFCloudAPI, workspaceId), nil)
 	if err != nil {
 		return nil, err
 	}
 
-	if res.StatusCode < 200 || res.StatusCode >= 400 {
-		return nil, errors.Errorf("error requesting terraform cloud backend state: status code: %d", res.StatusCode)
+	token := opts.TFCloudToken
+	if token == "" {
```

Comment:
I can totally do that in an unexported function but I got one question in regards to the implementation.

Since in the `Read` method we look if `t.reader == nil`, I can call this function just below the if statement to avoid recalling this function each time we Read bytes.

Like that I can remove the first 2 line to check for `Authorization` header.

Reply:
Yeah I think you can do that, I written the snippet quickly just for you to get the idea but it can be improved for sure